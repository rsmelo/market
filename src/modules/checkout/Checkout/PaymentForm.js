import React from 'react'
import { Field } from 'redux-form'

import InputField from '../../../components/InputField'
import normalizeCreditCard from '../../../utils/normalize/normalizeCreditCard'
import normalizeCardExpirationDate from '../../../utils/normalize/normalizeCardExpirationDate'
import style from './style.css'

const maskCardNumber = '0000 0000 0000 0000'
const maskCardExpirationDate = '00/00'

const PaymentForm = () => (
  <div className={style.form}>
    <h2>3. Pagamento</h2>
    <label htmlFor="cardNumber">Número do cartão</label>
    <Field
      id="cardNumber"
      name="cardNumber"
      type="input"
      normalize={normalizeCreditCard}
      component={InputField}
      placeholder={maskCardNumber}
    />
    <label htmlFor="cardHolderName">Nome do titular</label>
    <Field
      id="cardHolderName"
      name="cardHolderName"
      type="input"
      component={InputField}
    />
    <label htmlFor="cardExpirationDate">Expira em</label>
    <Field
      id="cardExpirationDate"
      name="cardExpirationDate"
      type="input"
      normalize={normalizeCardExpirationDate}
      component={InputField}
      placeholder={maskCardExpirationDate}
    />
    <label htmlFor="cardCvv">Código de segurança</label>
    <Field
      id="cardCvv"
      name="cardCvv"
      type="input"
      maxLength={4}
      component={InputField}
    />
  </div>
)

export default PaymentForm
