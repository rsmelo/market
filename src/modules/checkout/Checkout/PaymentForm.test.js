import React from 'react'
import renderer from 'react-test-renderer'

import PaymentForm from './PaymentForm'

jest.mock('redux-form', () => ({ Field: 'Field' }))

describe('PaymentForm', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<PaymentForm />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
