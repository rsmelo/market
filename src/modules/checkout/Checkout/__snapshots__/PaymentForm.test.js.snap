// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaymentForm renders correctly 1`] = `
<div
  className={undefined}
>
  <h2
    className={undefined}
  >
    3. Pagamento
  </h2>
  <label
    className={undefined}
    htmlFor="cardNumber"
  >
    Número do cartão
  </label>
  <Field
    component={[Function]}
    id="cardNumber"
    name="cardNumber"
    normalize={[Function]}
    placeholder="0000 0000 0000 0000"
    type="input"
  />
  <label
    className={undefined}
    htmlFor="cardHolderName"
  >
    Nome do titular
  </label>
  <Field
    component={[Function]}
    id="cardHolderName"
    name="cardHolderName"
    type="input"
  />
  <label
    className={undefined}
    htmlFor="cardExpirationDate"
  >
    Expira em
  </label>
  <Field
    component={[Function]}
    id="cardExpirationDate"
    name="cardExpirationDate"
    normalize={[Function]}
    placeholder="00/00"
    type="input"
  />
  <label
    className={undefined}
    htmlFor="cardCvv"
  >
    Código de segurança
  </label>
  <Field
    component={[Function]}
    id="cardCvv"
    maxLength={4}
    name="cardCvv"
    type="input"
  />
</div>
`;
